Source/Core/Core/PowerPC/PowerPC.h

```diff
@@ -162,6 +164,8 @@ void InjectExternalCPUCore(CPUCoreBase* core);
 // Stepping requires the CPU Execution lock (CPU::PauseAndLock or CPU Thread)
 // It's not threadsafe otherwise.
 void SingleStep();
+bool StepOut(std::optional<std::chrono::seconds> timeout = std::nullopt);
```

Comment:
Is there a realistic benefit to allowing no specified timeout value?

Reply:
That is, as opposed to requiring manually specifying zero to make intent clear.

Reply:
It will step until the function returns if no timeout is specified (i.e. will always succeed). I'll need this feature in another PR.

Reply:
That's probably something that should be documented above its prototype in the header file (stepping until it returns).

Reply:
Also, it's generally better to introduce behavioral changes (note, your PR title says 'move', which implies no modification to underlying behavior, yet here we are :p) like this in the PR that actually makes use of it, not in the preceding migration, since it gives more context to people reviewing your change.